sdk/javascript/test/utils/arrayHelpers_spec.js

```diff
@@ -0,0 +1,268 @@
+			const context = new TestContext([10, 11, 12, 13, 6]);
+
+			// Act:
+			const elements = arrayHelpers.readArray(context.subView, context.factory);
+
+			// Assert:
+			expect(elements).to.deep.equal(expectedElements);
+		});
+
+		it('can read when using accessor and elements are ordered', () => {
+			// Arrange:
+			const context = new TestContext([10, 11, 12, 13, 6]);
+
+			// Act:
+			const elements = arrayHelpers.readArray(context.subView, context.factory, element => element.tag);
+
+			// Assert:
+			expect(elements).to.deep.equal(expectedElements);
+		});
+
+		it('cannot read when using accessor and elements are not ordered', () => {
+			// Arrange:
+			const context = new TestContext([10, 11, 12, 13, 6]);
+
+			// Assert: use negative to fake wrong ordering
+			expect(() => arrayHelpers.readArray(context.subView, context.factory, element => -element.tag))
+				.to.throw('elements in array are not sorted');
+		});
+
+		assertThrowsWhenAnyElementHasZeroSize(arrayHelpers.readArray);
+	});
+
+	describe('readArrayCount', () => {
+		const expectedElements = [
+			{ size: 10, tag: 15 },
+			{ size: 11, tag: 25 },
+			{ size: 12, tag: 36 }
+		];
+
+		it('reads specified number of elements', () => {
+			// Arrange:
+			const context = new TestContext([10, 11, 12, 42, 78]);
+
+			// Act:
+			const elements = arrayHelpers.readArrayCount(context.subView, context.factory, 3);
+
+			// Assert:
+			expect(elements).to.deep.equal(expectedElements);
+		});
+
+		it('can read when using accessor and elements are ordered', () => {
+			// Arrange:
+			const context = new TestContext([10, 11, 12, 42, 78]);
+
+			// Act:
+			const elements = arrayHelpers.readArrayCount(context.subView, context.factory, 3, element => element.tag);
+
+			// Assert:
+			expect(elements).to.deep.equal(expectedElements);
+		});
+
+		it('cannot read when using accessor and elements are not ordered', () => {
+			// Arrange:
+			const context = new TestContext([10, 11, 12, 42, 78]);
+
+			// Assert: use negative to fake wrong ordering
+			expect(() => arrayHelpers.readArrayCount(context.subView, context.factory, 3, element => -element.tag))
+				.to.throw('elements in array are not sorted');
+		});
+
+		assertThrowsWhenAnyElementHasZeroSize((view, factory) => arrayHelpers.readArrayCount(view, factory, 3));
+	});
+
+	describe('readVariableSizeElements', () => {
+		it('reads all available elements', () => {
+			// Arrange: aligned sizes 8, 12, 12, 16, 4
+			const context = new TestContext([7, 11, 12, 13, 3]);
+			const expectedElements = [
+				{ size: 7, tag: 15 },
+				{ size: 11, tag: 15 + 8 },
+				{ size: 12, tag: 15 + 8 + 12 },
+				{ size: 13, tag: 15 + 8 + 12 + 12 },
+				{ size: 3, tag: 15 + 8 + 12 + 12 + 16 }
+			];
+
+			// Act:
+			const elements = arrayHelpers.readVariableSizeElements(context.subView, context.factory, 4);
+
+			// Assert:
+			expect(elements).to.deep.equal(expectedElements);
+		});
+
+		it('throws when reading would result in OOB read', () => {
```

Comment:
would prefer to split this into two tests, i.e. throws on OOB read, can read entire buffer

Reply:
can read entire is already covered (by "readArray reads all available elements"). redid a bit, maybe now it's clearer

Reply:
would at least use subscopes, so only one context is active at a time